vital/optional.h

```diff
@@ -9,991 +9,1275 @@
-#     define TR2_OPTIONAL_CLANG_3_4_2_AND_HIGHER_
-#   endif
+#if defined __clang_major__
+# if ( __clang_major__ == 3 && __clang_minor__ >= 5 )
+#  define TR2_OPTIONAL_CLANG_3_5_AND_HIGHTER_
+# elif ( __clang_major__ > 3 )
+#  define TR2_OPTIONAL_CLANG_3_5_AND_HIGHTER_
 # endif
+# if defined TR2_OPTIONAL_CLANG_3_5_AND_HIGHTER_
+#  define TR2_OPTIONAL_CLANG_3_4_2_AND_HIGHER_
+# elif ( __clang_major__ == 3 && __clang_minor__ == 4 && \
+         __clang_patchlevel__ >= 2 )
+#  define TR2_OPTIONAL_CLANG_3_4_2_AND_HIGHER_
+# endif
+#endif
 #
-# if defined _MSC_VER
-#   if (_MSC_VER >= 1900)
-#     define TR2_OPTIONAL_MSVC_2015_AND_HIGHER___
-#   endif
+#if defined _MSC_VER
+# if ( _MSC_VER >= 1900 )
+#  define TR2_OPTIONAL_MSVC_2015_AND_HIGHER___
 # endif
+#endif
 
-# if defined __clang__
-#   if (__clang_major__ > 2) || (__clang_major__ == 2) && (__clang_minor__ >= 9)
-#     define OPTIONAL_HAS_THIS_RVALUE_REFS 1
-#   else
-#     define OPTIONAL_HAS_THIS_RVALUE_REFS 0
-#   endif
-# elif defined TR2_OPTIONAL_GCC_4_8_1_AND_HIGHER___
-#   define OPTIONAL_HAS_THIS_RVALUE_REFS 1
-# elif defined TR2_OPTIONAL_MSVC_2015_AND_HIGHER___
-#   define OPTIONAL_HAS_THIS_RVALUE_REFS 1
+#if defined __clang__
+# if ( __clang_major__ > 2 ) || ( __clang_major__ == 2 ) && \
+  ( __clang_minor__ >= 9 )
+#  define OPTIONAL_HAS_THIS_RVALUE_REFS 1
 # else
-#   define OPTIONAL_HAS_THIS_RVALUE_REFS 0
+#  define OPTIONAL_HAS_THIS_RVALUE_REFS 0
 # endif
+#elif defined TR2_OPTIONAL_GCC_4_8_1_AND_HIGHER___
+# define OPTIONAL_HAS_THIS_RVALUE_REFS 1
+#elif defined TR2_OPTIONAL_MSVC_2015_AND_HIGHER___
+# define OPTIONAL_HAS_THIS_RVALUE_REFS 1
+#else
+# define OPTIONAL_HAS_THIS_RVALUE_REFS 0
+#endif
 
+#if defined TR2_OPTIONAL_GCC_4_8_1_AND_HIGHER___
+# define OPTIONAL_HAS_CONSTEXPR_INIT_LIST 1
+# define OPTIONAL_CONSTEXPR_INIT_LIST constexpr
+#else
+# define OPTIONAL_HAS_CONSTEXPR_INIT_LIST 0
+# define OPTIONAL_CONSTEXPR_INIT_LIST
+#endif
 
-# if defined TR2_OPTIONAL_GCC_4_8_1_AND_HIGHER___
-#   define OPTIONAL_HAS_CONSTEXPR_INIT_LIST 1
-#   define OPTIONAL_CONSTEXPR_INIT_LIST constexpr
-# else
-#   define OPTIONAL_HAS_CONSTEXPR_INIT_LIST 0
-#   define OPTIONAL_CONSTEXPR_INIT_LIST
-# endif
-
-# if defined TR2_OPTIONAL_CLANG_3_5_AND_HIGHTER_ && (defined __cplusplus) && (__cplusplus != 201103L)
-#   define OPTIONAL_HAS_MOVE_ACCESSORS 1
-# else
-#   define OPTIONAL_HAS_MOVE_ACCESSORS 0
-# endif
+#if defined TR2_OPTIONAL_CLANG_3_5_AND_HIGHTER_ && ( defined __cplusplus ) && \
+  ( __cplusplus != 201103L )
+# define OPTIONAL_HAS_MOVE_ACCESSORS 1
+#else
+# define OPTIONAL_HAS_MOVE_ACCESSORS 0
+#endif
 
-# // In C++11 constexpr implies const, so we need to make non-const members also non-constexpr
-# if (defined __cplusplus) && (__cplusplus == 201103L)
-#   define OPTIONAL_MUTABLE_CONSTEXPR
-# else
-#   define OPTIONAL_MUTABLE_CONSTEXPR constexpr
-# endif
+# \
+  // In C++11 constexpr implies const, so we need to make non-const members also non-constexpr
```

Comment:
```suggestion
// In C++11 constexpr implies const, so we need to make non-const members also non-constexpr
```
(I have no idea what was with the original `# //`.)